src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -505,6 +526,9 @@ public HashSet<(uint256, DateTimeOffset)> RemoveAllTransactions()
         {
             var removedTransactions = new HashSet<(uint256, DateTimeOffset)>();
 
+            if (!this.IsWalletActive())
```

Comment:
`RemoveAllTransactions` this is odd.... dont we need to update the sync height if we remove all trx?

Reply:
That's done by the code that calls this method. We don't want the side-effect here of activating the wallet or changing the height at which it will activate,